nailgun/entities.py

```diff
@@ -2166,6 +2200,59 @@ def update_payload(self, fields=None):
         """Wrap submitted data within an extra dict."""
         return {u'hostgroup': super(HostGroup, self).update_payload(fields)}
 
+    def path(self, which=None):
+        """Extend ``nailgun.entity_mixins.Entity.path``.
+        The format of the returned path depends on the value of ``which``:
+
+        puppetclass_ids
+            /api/hostgroups/:hostgroup_id/puppetclass_ids
+        smart_class_parameters
+            /api/puppetclasses/:puppetclass_id/smart_class_parameters
+
+        Otherwise, call ``super``.
+
+        """
+        if which in ('puppetclass_ids', 'smart_class_parameters'):
+            return '{0}/{1}'.format(
+                super(HostGroup, self).path(which='self'),
+                which
+            )
```

Comment:
Your docstring says `/api/puppetclasses/:puppetclass_id/smart_class_parameters`, but here you'll form `/api/hostgroups/:hostgroup_id/smart_class_parameters`. Which one is actually valid?